src/inmanta/protocol/ipc_light.py

```diff
@@ -246,3 +255,41 @@ def connection_lost(self, exc: Exception | None) -> None:
         super().connection_lost(exc)
         for fin in self.finalizers:
             asyncio.get_running_loop().create_task(fin())
+
+
+class LogReceiver(IPCFrameProtocol[ServerContext]):
+    """
+    IPC feature to receive log message
+    """
+
+    def frame_received(self, frame: IPCRequestFrame[ServerContext, ReturnType] | IPCReplyFrame) -> None:
+        if isinstance(frame, IPCLogRecord):
+            # calling log here is safe because if there are nu arguments, formatter is never called
+            logging.getLogger(frame.name).log(frame.levelno, frame.msg)
+        else:
+            super().frame_received(frame)
+
+
+class LogShipper(logging.Handler):
+    """
+    Log sender associated with the log receiver
+
+    This sender is threadsafe
+    """
+
+    def __init__(self, protocol: IPCFrameProtocol[object], eventloop: asyncio.BaseEventLoop) -> None:
+        self.protocol = protocol
+        self.eventloop = eventloop
+        super().__init__()
+
+    def emit(self, record: logging.LogRecord) -> None:
+        self.eventloop.call_soon_threadsafe(
```

Comment:
The failure behavior here is bit of an open question:

1. if the channel goes down
2. we raise a closed exception
3. that causes `call_soon_threadsafe`  to feed it to `self._loop.call_exception_handler`

Would it be better to silently drop log lines if we are disconnected?

Reply:
it will indeed terminate it, but logs produced during termination will be quite ugly (and invisible)

Reply:
it even logs the failure, so it is a loop. I'll update it